src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java

```diff
@@ -496,6 +496,38 @@ public boolean checkSlotOnPageLoaded(String slotName) {
       restoreDefaultImplicitWait();
     }
   }
+  /**
+   * Check if AdEngine loaded the ad web elements inside slot with mobile state
+   */
+  public boolean checkSlotOnPageLoaded(String slotName, Boolean isMobile) {
+    WebElement slot;
+
+    changeImplicitWait(250, TimeUnit.MILLISECONDS);
+
+    try {
+      String slotSelector = AdsContent.getSlotSelector(slotName);
+      triggerAdSlotWithMobileState(slotName, isMobile);
+
+      try {
+        slot = driver.findElement(By.cssSelector(slotSelector));
+      } catch (NoSuchElementException elementNotFound) {
+        Log.logError(String.format("Slot %s not found on the page", slotName), elementNotFound);
+
+        return false;
+      }
+
+      List<WebElement> adWebElements = slot.findElements(By.cssSelector("iframe"));
+
+      Log.log("Slot found",
+              String.format("%s found on the page with selector: %s", slotName, slotSelector),
+              true
+      );
+
+      return adWebElements.size() >= 1;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1155)